### ISSUE TITLE
Bug - BatchNormalization - normalization var and running var are not compatible

### ISSUE THREAD

**Issue by robertoalotufo** [pytorch/pytorch#3122]
After noticing the different behavior of a network with BatchNorm during training and eval(), I realized that the running_var is updated using **unbiased var** while the normalization var uses **biased var**:

File: `pytorch/torch/lib/THNN/generic/BatchNormalization.c`

`Line 40:         invstd = (real) (1 / sqrt(sum/n + eps));`

`Line 48:         accreal unbiased_var = sum / (n - 1);`

`invstd` (line 40) is used to normalize the mini-batch
`unbiased_var` (line 48) is used to update the `running_var` with `momentum` factor.

This can be corrected either by replacing `n` by `(n-1)` in line40 or replacing `(n-1)` by `n` in line48.

I have seen some complaints of other users regarding problems with batchnorm, difference from training and eval(). This may be the problem.

**Comment by ngimel** [comment]
Implementation follows batch normalization paper, https://arxiv.org/abs/1502.03167 that calls for unbiased variance when computing moving averages (page 4), and biased variance during training (page 3). It is expected that batch norm results are different during training and evaluation, because during training the batch is normalized with its own mean and (biased)  variance, and during evaluation accumulated running variance and mean are used, thus making evaluation batch-independent.
